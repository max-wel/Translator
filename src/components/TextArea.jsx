import React from 'react';
import PropTypes from 'prop-types';

const TextArea = props => {
  const { name, placeholder, value, onChange, readOnly } = props;
  return (
    <div className="form-group">
      <textarea
        className="form-control form-control-lg mb-3"
        name={name}
        rows="5"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        readOnly={readOnly}
      />
    </div>
  );
};

TextArea.defaultProps = {
  readOnly: false
};

TextArea.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  readOnly: PropTypes.bool
};

export default TextArea;
