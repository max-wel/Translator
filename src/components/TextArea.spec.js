import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from './TextArea';

const props = {
  name: '',
  placeholder: '',
  value: '',
  onChange: () => {}
};

describe('<TextArea />', () => {
  it('renders a <TextArea /> component', () => {
    const output = renderer.create(<TextArea {...props} />);
    expect(output).toMatchSnapshot();
  });
});
