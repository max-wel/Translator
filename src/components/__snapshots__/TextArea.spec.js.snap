// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> renders a <TextArea /> component 1`] = `
<div
  className="form-group"
>
  <textarea
    className="form-control form-control-lg mb-3"
    name=""
    onChange={[Function]}
    placeholder=""
    readOnly={false}
    rows="5"
    value=""
  />
</div>
`;
